packages/upload-api/src/blob/replicate.js

```diff
@@ -94,8 +94,19 @@ export const blobReplicateProvider = (context) => {
         }
       }
 
-      // TODO: support reducing the number of replicas?
-      const newReplicasCount = nb.replicas - activeReplicas.length
+      // Note: We +1 below to include the source blob, which is not recorded in
+      // the replicas table.
+      const newReplicasCount = nb.replicas - (activeReplicas.length + 1)
+
+      // TODO: support reducing the number of replicas
+      if (newReplicasCount < 0) {
+        return Server.error(
+          /** @type {API.ReplicationCountRangeError} */ ({
+            name: 'ReplicationCountRangeError',
+            message: 'reducing replica count not implemented',
+          })
+        )
+      }
 
       // lets allocate some replicas!
       if (newReplicasCount > 0) {
```

Comment:
What is expected to happen when `newReplicasCount == 0`? Is it worth bailing early when that is the case, or will `selectReplicationProviders` and remaining logic handle it? Do we think it's worth adding a test to `packages/upload-api/src/test/handlers/space/blob/replicate.js` for this case if one doesn't exist?